test/boulder-tools/hotfix_release.sh

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/env bash
+
+# -e Stops execution in the instance of a command or pipeline error.
+# -u Treat unset variables as an error and exit immediately.
+set -eu
+
+STATUS="FAILURE"
+RST=$(tput sgr0)
+RED=$(tput bold && tput setaf 1)
+GRN=$(tput bold && tput setaf 2)
+BLU=$(tput bold && tput setaf 4)
+
+function exit_msg() {
+  # complain to STDERR and exit with error
+  echo "${*}" >&2
+  exit 2
+}
+
+function print_outcome() {
+  if [ "${STATUS}" == SUCCESS ]
+  then
+    echo
+    echo "${GRN}${STATUS}${RST}"
+  else
+    echo
+    echo "${RED}${STATUS}${RST}"
+  fi
+}
+
+function print_heading() {
+  echo
+  echo "${BLU}${1}${RST}"
+}
+
+function not_yes_no() {
+    case "${1}" in
+      [yY][eE][sS]|[yY]) create_branch="yes" && return 1 ;;
+      [nN][oO]|[nN]) create_branch="no" && return 1 ;;
+      *) return 0 ;;
+    esac
+}
+
+function get_user_input() {
+    while [ -z "${create_branch}" ]; do
+        read -p "${1}" create_branch
+        if not_yes_no "${create_branch}" 
+        then
+            echo "Need [yes/no], got [${create_branch}]"
+            create_branch=""
+        fi
+    done
+}
+
+# On EXIT, trap and print outcome.
+trap "print_outcome" EXIT
+
+print_heading "Ensuring main branch is up to date"
+git fetch --all
+git checkout origin/main
+
+print_heading "Fetching details for the most recent release tag"
+latest_tag_sha=$(git show-ref --tags | tail -1 | awk '{print $1}')
+latest_tag_name=$(git show-ref --tags | tail -1 | awk '{print $2}' | sed 's|refs\/tags\/||')
+
+print_heading "Latest tag:"
+echo "${latest_tag_name}"
+echo "${latest_tag_sha}"
+
+
+print_heading "Hotfix release tag:"
+if [[ "${latest_tag_name: -1}" =~ ^[0-9] ]]
+then
+    new_tag_name="${latest_tag_name}a"
+else
+    next_tag_letter=$(echo "${latest_tag_name: -1}" |  tr "0-9a-z" "1-9a-z_")
+    tag_with_last_char_removed=$(echo "${latest_tag_name}" | sed 's|.$||')
+    new_tag_name="${tag_with_last_char_removed}${next_tag_letter}"   
+fi
+echo "${new_tag_name}"
+
+create_branch=""
+release_branch_name=$(echo "${new_tag_name}" | sed 's|release-|release-branch-|')
+get_user_input "Create hotfix release branch: ${release_branch_name}? "
+if [ "${create_branch}" = yes ]
+then
+    print_heading "Creating new branch ${release_branch_name} @ ${latest_tag_sha}"
+    git checkout -b "${release_branch_name}" "${latest_tag_sha}"
+    git push --set-upstream origin "${release_branch_name}"
+
+    cherry_pick_sha=""
+    while [ -z "${cherry_pick_sha}" ]; do
```

Comment:
Looks like this is designed to let you cherry-pick multiple commits onto the release branch? I like it! Two things I'd change:
1) take the list of commits on the command line instead of from stdin
2) only do the push and tag after the very last one

Reply:
At present this takes just one cherry-pick. I was mostly going off of experience, but I don't think I've ever had to do more than a single cherry-pick for hot-fixes I've done here. I could add some flags and opt parsing to this to enable what you're asking for in a follow-up commit. How does that sound?

Reply:
Ah, I misunderstood. If the goal is to only cherry-pick one commit, then there's no need to have this inside a while loop.

I like the idea of accepting multiple commits to cherry pick (for example: that `08-30` release branch had four commits cherry picked onto it but only two hotfix releases tagged on it). But it's certainly not necessary for v1 of this script.

Either way, I'd still prefer to take the sha1 on the command line rather than stdin so that hotfix release commands can be easily copy-pasted and reviewed for correctness before they're executed.